esphome/components/uart/uart_esp8266.cpp

```diff
@@ -49,15 +55,29 @@ void UARTComponent::setup() {
   // is 1 we still want to use Serial.
   SerialConfig config = static_cast<SerialConfig>(get_config());
 
-  if (this->tx_pin_.value_or(1) == 1 && this->rx_pin_.value_or(3) == 3) {
+  if (!esphome::uart::UARTComponent::serial0InUse && this->tx_pin_.value_or(1) == 1 &&
```

Comment:
You don't actually need a fully qualified name here:
```suggestion
  if (!UARTComponent::serial0InUse && this->tx_pin_.value_or(1) == 1 &&
```